packages/opentelemetry-propagator-jaeger/src/JaegerHttpTracePropagator.ts

```diff
@@ -55,31 +59,67 @@ export class JaegerHttpTracePropagator implements TextMapPropagator {
 
   inject(context: Context, carrier: unknown, setter: TextMapSetter) {
     const spanContext = getSpanContext(context);
-    if (!spanContext || isInstrumentationSuppressed(context)) return;
-
-    const traceFlags = `0${(spanContext.traceFlags || TraceFlags.NONE).toString(
-      16
-    )}`;
-
-    setter.set(
-      carrier,
-      this._jaegerTraceHeader,
-      `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
-    );
+    const baggage = getBaggage(context);
+    if (spanContext && isInstrumentationSuppressed(context) === false) {
+      const traceFlags = `0${(
+        spanContext.traceFlags || TraceFlags.NONE
+      ).toString(16)}`;
+
+      setter.set(
+        carrier,
+        this._jaegerTraceHeader,
+        `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
+      );
+    }
+
+    if (baggage) {
+      for (const [key, entry] of baggage.getAllEntries()) {
+        setter.set(
+          carrier,
+          `${UBER_BAGGAGE_HEADER_PREFIX}-${key}`,
```

Comment:
I think a similar comment like [there ](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/propagators/opentelemetry-propagator-ot-trace#baggage-notes) should be added.
`fields` API can't follow spec here